src/generic/C1_helper.h

```diff
@@ -0,0 +1,1092 @@
+       
+       // Set the boundary parametrisation data pointer for this node
+       Nodal_boundary_parametrisation_pt[j_node] = boundary_parametrisation_pt;
+       
+       // Set the coordinate of node j on this boundary
+       Boundary_coordinate_of_node[j_node] = boundary_coord_of_node[j];
+       
+       update_rotation_matrices();
+      } // end of loop over nodes in node_on_boundary [j]
+    } // end of set_nodal_boundary_parametrisation()
+   
+
+   
+   /// Update all rotation matrices (checks if they are needed unless flag is
+   /// true)
+    void update_rotation_matrices()
+    {
+      // [zdec] hard coded the three vertex nodes
+      unsigned n_vertex = 3;
+      // Loop over each vertex
+      for (unsigned j_node = 0; j_node < n_vertex; j_node++)
+      {
+        // If this node does not have a parametrisation (the pointer is still
+        // null) skip over it, otherwise we go on to fill out the rotation
+        // matrix
+        if (!nodal_boundary_parametrisation_pt(j_node))
+        {
+          continue;
+        }
+
+        // Initialise the two basis vectors and their jacobians
+        Vector<Vector<double>> bi(2, Vector<double>(2, 0.0));
+        Vector<DenseMatrix<double>> dbidx(2, DenseMatrix<double>(2, 2, 0.0));
+
+        // Our new coordinate system:
+        //     (l, s)=(normal component, tangent component)
+        // which we define in terms of basis vectors (rescaled)
+        //     ni=dxi/dl / |n|           <-- Jacobian col 1
+        //     ti=dxi/ds / |t|           <-- Jacobian col 2
+        // and their derivatives
+        //     dnidxj=d/dxj(dxi/dl / |n|) <-- Hessian `col' 1
+        //     dtidxj=d/dxj(dxi/ds / |t|) <-- Hessian `col' 2
+
+        // [zdec] we use i and j for brevity
+        // but it should be alpha & beta
+        // Need to write up how the transformation is done
+
+        // Storage for our basis and derivatives
+        Vector<double> ni(2, 0.0);
+        Vector<double> ti(2, 0.0);
+        Vector<double> dnids(2, 0.0);
+        Vector<double> dtids(2, 0.0);
+
+        // All tensors assumed evaluated on the boundary
+        // Jacobian of inverse mapping
+        DenseMatrix<double> jac_inv(2, 2, 0.0);
+        // Hessian of mapping [zdec] (not needed because...)
+        Vector<DenseMatrix<double>> hess(2, DenseMatrix<double>(2, 2, 0.0));
+        // Hessian of inverse mapping [zdec] (...this can be found by
+        // hand)
+        Vector<DenseMatrix<double>> hess_inv(2, DenseMatrix<double>(2, 2, 0.0));
+
+        // The basis is defined in terms of the boundary parametrisation
+        Vector<double> boundary_coord = {Boundary_coordinate_of_node[j_node]};
+        C1CurviLine* boundary_pt =
+          Nodal_boundary_parametrisation_pt[j_node];
+        Vector<double> x(2, 0.0);
+        Vector<double> dxids(2, 0.0);
+        Vector<double> d2xids2(2, 0.0);
+
+        // Get position (debug) // hierher Aidan why debug?
```

Comment:
Position is not needed for the maths

Reply:
aidan will tidy when getting rid of debug (and [zdec])

Reply:
Keeping this unresolved for now to alert Aidan to the need to grep for "debug" and "[zdec]" when it's his turn to tidy.